lib/features/offline_mode/work_manager/sending_email_worker.dart

```diff
@@ -5,69 +5,68 @@ import 'package:core/presentation/state/success.dart';
 import 'package:tmail_ui_user/features/login/domain/state/get_stored_token_oidc_state.dart';
 import 'package:tmail_ui_user/features/login/domain/usecases/get_authenticated_account_interactor.dart';
 import 'package:tmail_ui_user/features/mailbox/domain/model/create_new_mailbox_request.dart';
-import 'package:tmail_ui_user/features/mailbox_dashboard/presentation/bindings/mailbox_dashboard_bindings.dart';
 import 'package:tmail_ui_user/features/offline_mode/bindings/sending_email_interactor_bindings.dart';
-import 'package:tmail_ui_user/features/offline_mode/exceptions/workmanager_exception.dart';
 import 'package:tmail_ui_user/features/offline_mode/manager/sending_email_cache_manager.dart';
 import 'package:tmail_ui_user/features/offline_mode/model/sending_state.dart';
-import 'package:tmail_ui_user/features/offline_mode/observer/work_observer.dart';
+import 'package:tmail_ui_user/features/offline_mode/work_manager/worker.dart';
 import 'package:tmail_ui_user/features/sending_queue/domain/extensions/sending_email_extension.dart';
 import 'package:tmail_ui_user/features/sending_queue/domain/model/sending_email.dart';
-import 'package:tmail_ui_user/features/sending_queue/domain/state/update_sending_email_state.dart';
-import 'package:tmail_ui_user/features/sending_queue/domain/usecases/delete_sending_email_interactor.dart';
-import 'package:tmail_ui_user/features/sending_queue/domain/usecases/update_sending_email_interactor.dart';
+import 'package:tmail_ui_user/features/sending_queue/presentation/bindings/sending_queue_bindings.dart';
+import 'package:tmail_ui_user/features/sending_queue/presentation/bindings/sending_queue_interactor_bindings.dart';
 import 'package:tmail_ui_user/features/sending_queue/presentation/utils/sending_queue_isolate_manager.dart';
 import 'package:tmail_ui_user/features/session/domain/extensions/session_extensions.dart';
 import 'package:tmail_ui_user/features/session/domain/state/get_session_state.dart';
 import 'package:tmail_ui_user/features/session/domain/usecases/get_session_interactor.dart';
 import 'package:tmail_ui_user/main/bindings/main_bindings.dart';
 import 'package:tmail_ui_user/main/routes/route_navigation.dart';
 
-class SendingEmailObserver extends WorkObserver {
-
-  AccountId? _currentAccountId;
-  Session? _currentSession;
-  SendingEmail? _sendingEmail;
+class SendingEmailWorker extends Worker {
 
   SendEmailInteractor? _sendEmailInteractor;
   GetAuthenticatedAccountInteractor? _getAuthenticatedAccountInteractor;
   DynamicUrlInterceptors? _dynamicUrlInterceptors;
   AuthorizationInterceptors? _authorizationInterceptors;
   GetSessionInteractor? _getSessionInteractor;
-  DeleteSendingEmailInteractor? _deleteSendingEmailInteractor;
   SendingQueueIsolateManager? _sendingQueueIsolateManager;
-  UpdateSendingEmailInteractor? _updateSendingEmailInteractor;
   SendingEmailCacheManager? _sendingEmailCacheManager;
 
-  Completer<bool>? _completer;
+  late Completer<bool> _completer;
```

Comment:
Don't you think use `late` here is dangerous ? IMO, use nullable and then when you surely assign to it, use with !

Reply:
In this case it is not dangerous at all. If you read carefully you can see that `Completer` used here must be a non-nullable variable. I can initialize it directly at declaration but I use `late` so that it is only assigned value during runtime, not compilation